sql/src/main/java/io/crate/planner/node/dql/RoutedCollectPhase.java

```diff
@@ -306,6 +306,7 @@ public RoutedCollectPhase normalize(EvaluatingNormalizer normalizer, Transaction
                 newWhereClause,
                 distributionInfo
             );
+            result.orderBy(orderBy);
```

Comment:
afaik we should patch this against master. also I think it was not related to unions at all but just occurred while developing. would be also interesting why this never caused issues on master